scripts/modules/ado/ado.psm1

```diff
@@ -278,6 +278,120 @@ Function Set-ServiceEndpoint() {
     }    
 }
 
+<#
+.SYNOPSIS
+Create an Azure Federated service endpoint (ServiceConnection).
+
+.DESCRIPTION
+Create an Azure Federated service endpoint (ServiceConnection).
+
+.PARAMETER  FederatedEndpointJsonPath
+Mandatory. Federated Service connection json file
+
+.PARAMETER ProjectName
+Mandatory. Azure devops project name
+
+.PARAMETER OrgnizationUri
+Mandatory. Azure devops project Orgnization Uri
+
+.EXAMPLE
+.\Set-FederatedServiceEndpoint -ArmServiceConnection <ArmServiceConnectionObject> -FederatedEndpointJsonPath <FederatedEndpointJsonPath> -ProjectName <ProjectName> OrgnizationUri <OrgnizationUri>
+#> 
+Function Set-FederatedServiceEndpoint() {
+    [CmdletBinding()]
+    Param(
+        [ValidateNotNullOrEmpty()]
+        [Parameter(ValueFromPipeline = $true)]
+        [Object]$ArmServiceConnection,      
+        [Parameter(Mandatory)]
+        [string]$FederatedEndpointJsonPath,              
+        [Parameter(Mandatory)]        
+        [string]$ProjectName,
+        [Parameter(Mandatory)]
+        [string]$OrgnizationUri
+    )
+
+    begin {
+        [string]$functionName = $MyInvocation.MyCommand    
+        Write-Debug "${functionName}:Entered"       
+        Write-Debug "${functionName}:FederatedEndpointJsonPath=$FederatedEndpointJsonPath"
+        Write-Debug "${functionName}:ProjectName=$ProjectName"
+        Write-Debug "${functionName}:OrgnizationUri=$OrgnizationUri"     
+    }
+
+    process {   
+        
+        Write-Debug "${functionName}:ArmServiceConnection=$($ArmServiceConnection | ConvertTo-Json -Depth 10)"
+        # Create Federated Identity Credential   
+
+        $appReg = Get-AzADApplication -DisplayName $ArmServiceConnection.appRegName   
+
+        $federatedCredentials = Get-AzADAppFederatedCredential -ApplicationObjectId $appReg.id
+        $federatedCredentials | Select-Object -Property Name
+
+        $devopsOrganizationName = $OrgnizationUri.substring(22)
```

Comment:
can this be renamed? it is confusing with next line

Reply:
renamed